### PR TITLE
Add detailed build instructions in Build.md

### PR DESCRIPTION
**Pull request by mhk**
These instructions explain how to build neovim-qt on Ubuntu 18.04 and
Windows 10 using MinWG and Visual Studio 2019.

### CODE REVIEW

File: Build.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    $ cd neovim-qt
+
+or download the zip file.
+
+    $ sudo apt install -y wget unzip
+    $ wget -c https://github.com/equalsraf/neovim-qt/archive/master.zip
+    $ unzip master.zip
+    $ cd neovim-qt-master
+
+Navigate to the `neovim-qt` source code directory and install the necessary dependencies (Qt, `cmake, compiler and `neovim`).
```
Comment (jgehrig) [inline Build.md]: Backtick typo on this line?

```
`Qt`, `cmake`, `compiler` and `neovim`
```

File: Build.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    > Invoke-WebRequest -Uri https://github.com/equalsraf/neovim-qt/archive/master.zip -OutFile master.zip
+    > Expand-Archive -Path master.zip .
+    > cd neovim-qt-master
+
+The easiest way to build `neovim-qt` on Windows 10 is to use the Qt installer (for example the [community edition](https://www.qt.io/download-open-source)).
+Download and run the Qt installer when asked which components to install select:
+
+- Qt _Version_
+    - MinGW _MinGW Version_
```
Comment (jgehrig) [inline Build.md]: super nit: A little confusing, the packages is written twice... Elsewhere too.

Maybe: `_{Version}_`?

File: Build.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    > Invoke-WebRequest -Uri https://github.com/equalsraf/neovim-qt/archive/master.zip -OutFile master.zip
+    > Expand-Archive -Path master.zip .
+    > cd neovim-qt-master
+
+After the successfull installation of Qt open a terminal (for example `powershell`) and navigate to the `neovim-qt` source code. 
+
+    > mkdir build
+    > cd build
+    > $env:Path += ";C:\Qt\Tools\CMake_64\bin\;C:\Qt\5.12.6\msvc2017_64\bin\;C:\Program Files\Neovim\bin"
```
Comment (jgehrig) [inline Build.md]: nit: You could expand this to make the directions below easier to follow/explain.

For example in cmd (pardon my lack of powershell):
```
> set CMAKE_PATH=C:\Qt\Tools\CMake_64\bin
> set MSVC_PATH=C:\Qt\5.12.6\msvc2017_64\bin
> set NVIM_PATH=C:\Program Files\Neovim\bin

> set PATH=%PATH%;%CMAKE_PATH%;%MSVC_PATH%...
```

**Comment by jgehrig** [comment]
Looks good! I added a few minor comments.

I think this would be helpful for everyone... I've seen a few requests like for more detailed build instructions.

Issue #547 mentioned adding instructions for MacOS, we should add a section (I would be happy to help/write).

**Comment by jgehrig** [comment]
I think the [Wiki](https://github.com/equalsraf/neovim-qt/wiki) would be a good place for these instructions? You would still want to keep the link in `README.md`...

File: Build.md (hunk 1 above)
Comment (mhk) [reply Build.md]: I tried to use the backticks for program names only.

File: Build.md (hunk 2 above)
Comment (mhk) [reply Build.md]: I thought about that, but decided initially against it to make clear, that the _MinGW Version_ can be different from the _CMake Version_. If you think this is more confusing than helping I will remove it.

File: Build.md (hunk 3 above)
Comment (mhk) [reply Build.md]: I like that. I updated the instructions accordingly.

**Comment by mhk** [comment]
It's your repo if you would like to create a wiki page that's fine by me.
I added it as markdown file because this way, the (detailed) build instructions will always be included in the repo when you download it.

File: Build.md (hunk 1 above)
Comment (jgehrig) [reply Build.md]: Sure, but I think there is still a typo :)

```
(Qt, `cmake`, compiler and `neovim`)
```

File: Build.md (hunk 2 above)
Comment (jgehrig) [reply Build.md]: Sure, sounds good.

**Comment by jgehrig** [comment]
> Btw: I do not have access to a MacOS, that's why I only included ubuntu and windows.

No problem! I have access to a VM, and can write something for MacOS.

> I added it as markdown file because this way, the (detailed) build instructions will always be included in the repo when you download it.

The markdown would remain exactly the same, but it would be submitted to `https://github.com/equalsraf/neovim-qt.wiki.git` instead.

IMO: There are basic instructions already in README. The Wiki makes edits/contributions easier, and has the advantage of not tracking unrelated binary assets in the main repo. We wouldn't need to add `pics/...`, which are not directly required for the build.

Ultimately, it's up to equalsraf.

**Comment by jgehrig** [comment]
Here are some basic MacOS build instructions. Feel free to tweak the language so that it fits with your Ubuntu/Windows instructions.

```
## MacOS

You must be logged in with an Apple ID, and on a recent version of MacOS supporting Xcode.

Open the `App Store`. Search for "Xcode", and install the latest version of. You may be prompted for log-in credentials.

You will need to install the `brew` package manager. Brew will also trigger the install of "Xcode Command Line Tools". See the [Brew Website](https://brew.sh) for more information.

Run the following command:

	$ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"


You will need to download the `neovim-qt` source code via Git:

    $ git clone https://github.com/equalsraf/neovim-qt.git
    $ cd neovim-qt

Alternatively, you can download a zip file:

	$ brew install wget
    $ wget -c https://github.com/equalsraf/neovim-qt/archive/master.zip
    $ unzip master.zip
    $ cd neovim-qt-master

Navigate to the `neovim-qt` source code directory and install the necessary dependencies:

	$ brew install neovim qt cmake
    $ mkdir build
    $ cd build
    $ cmake -DCMAKE_BUILD_TYPE=Release ..
    $ cmake --build .

If the build was successful, you can start `neovim-qt`.

    $ set NVIM_QT_RUNTIME_PATH=../src/gui/runtime bin/nvim-qt.app/Contents/MacOS/nvim-qt
```

File: Build.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    > $CMAKE_PATH = "C:\Qt\Tools\CMake_64\bin\"
+    > $QT_PATH = "C:\Qt\5.12.6\msvc2017_64\bin\"
+    > $NVIM_PATH = "C:\Program Files\Neovim\bin"
+    > $env:Path += "$CMAKE_PATH;$QT_PATH;$NVIM_PATH;"
+    > cmake -DCMAKE_PREFIX_PATH="C:\Qt\5.12.6\msvc2017_64\lib\cmake" -G "Visual Studio 16 2019" -DCMAKE_BUILD_TYPE=Release ..
+    > cmake --build .
+
+(These instructions assume the default installation path _C:\Qt_ for Qt, and that the `neovim` binary is in _C:\Program Files\Neovim\bin_.)
+
+If the build was successfull, you can start `neovim-qt`.
```
Comment (jgehrig) [inline Build.md]: nit spelling: successfull -> successful

**Comment by equalsraf** [comment]
> I think the [Wiki](https://github.com/equalsraf/neovim-qt/wiki) would be a good place for these instructions?
> 
> You would still want to keep the link in `README.md`...

Definitely wiki, here is the link https://github.com/equalsraf/neovim-qt/wiki/Build.

Thanks @mhk, I took the liberty of copying the contents of your Build.md into that wiki page.

Anyone should be able to edit the wiki, this way we dont need to go through PR whenever someone wants to had instructions for their favorite OS flavor.

Other than that a link in README should suffice.

**Comment by jgehrig** [comment]
Thanks @mhk! This is very helpful... This all looks good to me.

The Wiki may get more traffic if the link is at the top of the page. Just an idea; no strong opinions on where the link goes.

@equalsraf The link is updated and ready to merge.

**Comment by equalsraf** [comment]
Merged, thanks.

**Pull request merged**